### DISCUSSION TITLE
finplot performance

### DISCUSSION THREAD

**Discussion by idealisticLie** [highfestiva/finplot#429] [General]
Nice work, 

I know this was not designed for real-time candlestick plotting, but so fourth this is a great choice for something simple. 

I am scraping live price tick changes & plotting candlesticks which works well.
When using timer_callback on 1000ms callback there is a slight latency with application;

Do you know of any performance adjustments, or even rendering each candlestick differently? I use macOS.

thx

**Comment by highfestiva** [comment]
Without code, hard to tell. Make sure you perform any HTTP calls or similar outside of the timer callback. Instead do those first in a separate thread, and then have the timer callback update. Slight delays are currently not possible to circumvent, but if they're longer than those in `examples/complex.py`, you're either a) showing a lot of data, or b) doing something else slow in the callback.

**Reply by idealisticLie** [reply]
Performing requests outside seems to have worked. 

Though now it refreshes the zoom to max when calling back.

**Comment by highfestiva** [comment]
See `examples/complicated.py`.
